pandas/core/indexes/period.py

```diff
@@ -767,8 +768,8 @@ def _assert_can_do_setop(self, other):
             msg = DIFFERENT_FREQ_INDEX.format(self.freqstr, other.freqstr)
             raise IncompatibleFrequency(msg)
 
-    def _wrap_union_result(self, other, result):
-        name = self.name if self.name == other.name else None
+    def _wrap_setop_result(self, other, result):
+        name = get_op_result_name(self, other)
```

Comment:
I think could just remove this entirely here if ``_wrap_setop_result`` is defined in ``Index.base`` AND you move ``_apply_meta`` functionaily to inside of ``_shallow_copy`` for PI. (could also do this as a followup), but want to try to avoid duplicating ``_wrap_setup_result`` everywhere.

Reply:
The reason we can't move it to base from category is because in the case of category, we need to preserve the categories, and in the case of period, we need to preserve frequency. 

So we could investigate in a separate PR about moving the `_apply_meta()` functionality.

Reply:
yeah that's fine